src/qml/WiFiConnectionDialog.qml

```diff
@@ -0,0 +1,208 @@
+                                    "requirement": fields[key]["Requirement"]
+                                })
+                console.log(key + ":")
+                for (var inkey in fields[key]) {
+                    console.log("    " + inkey + ": " + fields[key][inkey])
+                }
+            }
+            userAgent.sendUserReply({"Passphrase": passphraseField.text})
+        }
+
+        onErrorReported: {
+            console.log("Got error from model: " + error)
+            failDialog.subLabelText = error
+            failDialog.open()
+        }
+    }
+    Connections {
+        target: modelData
+        function onConnectRequestFailed(error) {
+            console.log(error)
+        }
+
+        function onConnectedChanged(connected) {
+            if(connected) {
+                layerStack.pop(layerStack.currentLayer)
+            }
+        }
+    }
+    Flickable {
+        anchors.fill: parent
+        contentHeight: contentColumn.implicitHeight
+        Column {
+            id: contentColumn
+            width: parent.width
+            Item {height: Dims.h(15); width: parent.width}
+            Marquee {
+                anchors {
+                    left: parent.left
+                    right: parent.right
+                    leftMargin: dialogItem.rowMargin
+                    rightMargin: dialogItem.rowMargin
+                }
+                text: modelData.name
+                font.pixelSize: Dims.l(6)
+                height: Dims.l(8)
+            }
+            Column {
+                id: loginFieldsColumn
+                visible: !(modelData.connected || modelData.favorite)
+                anchors {
+                    left: parent.left
+                    right: parent.right
+                    leftMargin: dialogItem.rowMargin
+                    rightMargin: dialogItem.rowMargin
+                }
+                Label {
+                    id: identityLabel
+                    text: qsTrId("id-wifi-login")+":"
+                    font.pixelSize: Dims.l(6)
+                    visible: modelData.securityType === NetworkService.SecurityIEEE802
+                }
+
+                TextField {
+                    id: identityField
+                    text: modelData.identity
+                    width: parent.width
+                    visible: modelData.securityType === NetworkService.SecurityIEEE802
+                }
+
+                Label {
+                    id: passphraseLabel
+                    text: qsTrId("id-wifi-password")+":"
+                    font.pixelSize: Dims.l(6)
+                    visible: !(modelData.securityType == NetworkService.SecurityNone)
+                }
+
+                TextField {
+                    id: passphraseField
+                    text: modelData.passphrase
+                    echoMode: TextInput.Password //smartwatches are hard to type on. it is worth adding a 'show password' button for this field
```

Comment:
Delegating all UX decisions to @eLtMosen but I don't think we gain very much from hiding the password here. Imho these password placeholders are for situation where the user is sitting in a crowded space and types a password that they don't want anyone around to see. I don't really see this happening with a user typing their Wi-Fi passphrase once on their watch. I don't feel particularly strongly either way but this echoMode feels a bit overkill to me.